node/bft/src/helpers/pending.rs

```diff
@@ -146,29 +160,55 @@ impl<T: Copy + Clone + PartialEq + Eq + Hash, V: Clone> Pending<T, V> {
     /// If the `item` does not exist, `None` is returned.
     pub fn remove(&self, item: impl Into<T>, callback_value: Option<V>) -> Option<HashSet<SocketAddr>> {
         let item = item.into();
-        // Remove the item from the pending queue.
-        let result = self.pending.write().remove(&item);
-        // Remove the callback for the item, and process any remaining callbacks.
-        if let Some(callbacks) = self.callbacks.lock().remove(&item) {
-            if let Some(callback_value) = callback_value {
-                // Send a notification to the callback.
-                for (callback, _, _) in callbacks {
-                    callback.send(callback_value.clone()).ok();
+        // Remove the item from the pending queue and process any remaining callbacks.
+        match self.pending.write().remove(&item) {
+            Some(callbacks) => {
+                // Get the peer IPs.
+                let peer_ips = callbacks.keys().cloned().collect();
```

Comment:
```suggestion
                let peer_ips = callbacks.keys().copied().collect();
```
since `SocketAddr` is `Copy`